### PR TITLE
support tunnel MVP

### PR DESCRIPTION
**Pull request by rtertiaer**
### What does this change intend to accomplish?
This PR brings in [our support tunnel software](https://github.com/micro-nova/support_tunnel) in a pretty MVP fashion. It has an interface in the updater to permit requesting a support tunnel. It also implements two commands that are run via cronjob - one that reaps/garbage collects completed or expired tunnel sessions, and one that checks if any requested tunnels have launched in the remote side.

I'm submitting now for review, though I still need to check a couple of the checklist items off the list. I'd also like to produce a write-up describing this stack before landing it in a release, but I'd like to do that soon too. 

### Checklist

* [x] Have you tested your changes and ensured they work?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] If applicable, have you updated the documentation/manual?
* [x] If applicable, have you updated the CHANGELOG?
* [x] Does your submission pass linting & tests? You can test on localhost using `./scripts/test`
* [ ] Have you written new tests for your core features/changes, as applicable?
* [x] If this is a UI change, have you tested it across multiple browser platforms?
* [x] If this is a UI change, have you tested across multiple viewport sizes (ie. desktop versus mobile)?

### CODE REVIEW

File: amplipi/updater/static/index.html [hunk 1]
```diff
@@ -84,6 +87,20 @@ <h1><span class="text-white">Ampli</span><span class="text-danger">Pi</span></h1
...
+                This dialog allows you to request a support tunnel. Once requested, you must give the tunnel ID and preshared key to support, who can be reached via <a href="mailto:support@micro-nova.com">support@micro-nova.com</a>.
+              </div>
+              <div id="support-tunnel-button">
+                <a id="request-support-tunnel" role="button" class="btn btn-lg btn-primary text-center mt-3" onClick="requestSupportTunnel();">Request support tunnel</a>
+              </div>
+              <div id="support-tunnel-detail-container" class="d-none">
+                <pre id="support-tunnel-detail" class="alert alert-dark mt-3"></pre>
+                <div id="support-tunnel-detail-caption">
+                    Copy & paste the above lines in a support request.
+                </div>
```
Comment (rtertiaer) [inline amplipi/updater/static/index.html:101]: Unfortunately, the "copy to clipboard" mechanisms in Javascript are only available within secure contexts - localhost, and things over HTTPS. We are neither.

File: scripts/configure.py [hunk 2]
```diff
@@ -114,6 +114,32 @@
       'sudo rm /var/log/user*     && echo "removed user logs"   || echo ok',
     ]
   },
+  'support_tunnel': {
+    'apt': [
+      'libsystemd-dev'
```
Comment (rtertiaer) [inline scripts/configure.py:119]: allows logging directly to systemd

File: amplipi/updater/static/index.html [hunk 3]
```diff
@@ -84,6 +87,20 @@ <h1><span class="text-white">Ampli</span><span class="text-danger">Pi</span></h1
               </div>
               <a id="password" role="button" class="btn btn-lg btn-primary text-center mt-3" onClick="set_password();">Update Password</a>
             </div>
+            <div class="tab-pane fade text-left" id="support-tunnel" role="tabpanel" aria-labelledby="support-tunnel-tab">
+              <div id="support-tunnel-dialog">
+                This dialog allows you to request a support tunnel. Once requested, you must give the tunnel ID and preshared key to support, who can be reached via <a href="mailto:support@micro-nova.com">support@micro-nova.com</a>.
```
Comment (linknum23) [inline amplipi/updater/static/index.html:92]: Any chance we can embed the key in this uri?

href="mailto:support@micro-nova.com?subject=Support%20Tunnel%20Request&body=KEY"

File: scripts/configure.py [hunk 4]
```diff
@@ -114,6 +114,32 @@
...
+      'sudo mkdir -p /var/lib/support_tunnel',
+      'sudo chmod 0777 /var/lib/support_tunnel', # TODO: lock this down
+      'if [ ! -e /opt/support_tunnel ] ; then'
+      '  pushd $(mktemp --directory)',
+      '  git clone https://github.com/micro-nova/support_tunnel.git',
+      '  sudo mv support_tunnel /opt',
+      '  popd',
+      'fi',
+      'pushd /opt/support_tunnel',
+      'git pull',
```
Comment (linknum23) [inline scripts/configure.py:135]: What happens when we rewrite history? Maybe this is a little safer? `git fetch && git reset --hard main` I don't know of a better way to get latest safely.

File: scripts/configure.py (hunk 2 above)
Comment (linknum23) [reply scripts/configure.py:119]: might as well add this as a comment

File: amplipi/updater/static/index.html (hunk 3 above)
Comment (linknum23) [reply amplipi/updater/static/index.html:92]: If thats a bigger lift we can save it for later.

**Comment by linknum23** [comment]
Trying this out we found a couple of issues

- [ ] Needs .env file
- [ ] Add ssh-key in gcloud
- [ ] button could use spinner
- [ ] user needs to setup default;t creds in gcloud outside of init?
- [ ] fab get XXX should return json not python dict
- [ ] no wiregaurd dir

**Review by linknum23 — APPROVED** [review]
Looks good. Nice work Ryan!

**Comment by rtertiaer** [comment]
This latest rendition is running on a customer's appliance in the field and works a charm. I'm gonna let it run for a day or two but I feel pretty comfortable with this rendition making it into a release.

File: pyproject.toml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "amplipi"
-version = "0.3.5"
+version = "0.3.5+7d6801f-support_tunnel-dirty"
```
Comment (linknum23) [inline pyproject.toml:3]: oops

**Review by linknum23 — APPROVED** [review]

**Pull request merged**